flow/project.py

```diff
@@ -4593,20 +4602,26 @@ def foo(job):
         return group_entry
 
     def _register_groups(self):
-        """Register all groups and add the correct operations to each."""
+        """Register all groups.
+
+        Operations are assigned to each group.
+
+        Aggregators are created for each group and tracked in a bidirectional mapping.
+
+        """
         group_entries = []
         # Gather all groups from class and parent classes.
         for cls in type(self).__mro__:
             group_entries.extend(getattr(cls, "_GROUPS", []))
 
-        aggregators = defaultdict(list)
-        # Initialize all groups without operations.
-        # Also store the aggregates we need to store all the groups associated
-        # with each aggregator.
+        # Initialize all groups without operations. Also store the aggregators
+        # associated with each group.
         for entry in group_entries:
-            self._groups[entry.name] = FlowGroup(entry.name, options=entry.options)
-            aggregators[entry.aggregator].append(entry.name)
-        self._aggregator_per_group = dict(aggregators)
+            group = FlowGroup(entry.name, options=entry.options)
+            self._groups[entry.name] = group
+            self._group_to_aggregator[group] = entry.aggregator._create_AggregatesStore(
+                self
+            )
```

Comment:
Here we aren't checking any more that any repeated aggregate stores aren't created. We should check first to see if the entry's aggregate already exists in the bimap.

Reply:
Good observation. Thanks to @b-butler and @kidrahahjo for pointing this out. Fixed in c0be467f615f5559fa88eb1d1761ee498ffa0198.